lib/subscriptions_test_kit/suites/subscriptions_r5_backport_r4_server/subscription_rejection/reject_subscription_filter_test.rb

```diff
@@ -0,0 +1,79 @@
+        defined in the requested topic and are implemented in the server.
+
+        The test will pass if the server either
+        1. rejects the Subscription by responding with a non-201 response, or
+        2. updates the Subscription resource to remove or replace the unsupported value.
+      )
+
+      verifies_requirements 'hl7.fhir.uv.subscriptions_1.1.0@9'
+
+      input :subscription_resource,
+            title: 'Workflow Subscription Resource',
+            type: 'textarea',
+            description: %(
+              A Subscription resource in JSON format that Inferno will send to the server under test
+              so that it can demonstrate its ability to perform the Subscription creation and Notification
+              response workflow. The instance must be conformant to the R4/B Topic-Based Subscription profile.
+              Inferno may modify the Subscription before submission, e.g., to point to Inferno's notification endpoint.
+              This input is also used by the unsupported Subscription test as the base on which to add unsupported
+              element values to test for server rejection.
+            )
+      input :unsupported_subscription_filter,
+            title: 'Unsupported Subscription Filter',
+            description: %(A value for `filterCriteria` extension under the `criteria` that is not implemented by the
+                           server to test for Subscription rejection.),
+            optional: true
+
+      run do
+        assert_valid_json(subscription_resource)
+        subscription = JSON.parse(subscription_resource)
+
+        unsupported_info = {
+          'unsupported_title' => 'unsupported filter criteria',
+          'field_path' => ['_criteria'],
+          'field_value' => if unsupported_subscription_filter.nil?
+                             unsupported_subscription_filter
+                           else
+                             { 'extension' => [{
+                               url: 'http://hl7.org/fhir/uv/subscriptions-backport/StructureDefinition/backport-filter-criteria',
+                               valueString: unsupported_subscription_filter
+                             }] }
+                           end
+        }
+
+        skip_if(unsupported_info['field_value'].blank?, %(
+          No subscription filter input provided.))
+
+        field_name = unsupported_info['field_path'].last
+
+        outer_field_name = unsupported_info['field_path'].first
+        subscription_field = if unsupported_info['field_path'].length > 1
+                               subscription[outer_field_name]
+                             else
+                               subscription
+                             end
+
+        subscription_field[field_name] = unsupported_info['field_value']
+
+        send_unsupported_subscription(subscription, unsupported_info['unsupported_title'],
```

Comment:
This logic doesn't seem to be working either - when run against the client tests that echo back the bad value and return 201 created, the logic passes it when I think it should be failing.

Reply:
Addressed in [06d86cb](https://github.com/inferno-framework/subscriptions-test-kit/pull/7/commits/06d86cb11ecdc85f1d8d5b32c62be7cf682edc91). Made changes to `send_unsupported_subscription` to normalize hash keys before comparing.